src/core/fonts.js

```diff
@@ -1905,8 +1840,6 @@ var Font = (function FontClosure() {
           glyf.data.set(newGlyfData.subarray(0, firstEntryLength), writeOffset);
           itemEncode(loca.data, locaData.length - itemSize,
                      writeOffset + firstEntryLength);
-        } else {
-          glyf.data = newGlyfData.subarray(0, writeOffset);
```

Comment:
I've waited too long to look at this again and now I'm not exactly sure. I think at some point I was going to get rid of the dupFirstEntry option since it is pretty much always true unless there's not room to duplicate the glyph. Let me try adding it back.